toss/fitness/fitness_function_utils.py

```diff
@@ -143,25 +117,140 @@ def compute_space_coverage(number_of_spacecrafts: int, spin_axis: np.ndarray, sp
+        theta (np.ndarray): Array of theta coordinates for each point defined on the spherical tensor.
+        phi (np.ndarray): Array of phi coordinates for each point defined on the spherical tensor.
+        bool_tensor (np.ndarray): Boolean tensor corresponding to each point defined on the spherical grid.
+
+    Returns:
+        bool_tensor (np.ndarray): Updated boolean tensor corresponding to each point defined on the spherical grid.
+    """
+    # Rotate positions according to body's rotation to simulate that the grid (i.e gravitational field approximation) is also rotating accrdingly
+    rotated_positions = None
+    
+    pos = np.array_split(positions, number_of_spacecrafts, axis=1)
+    for counter, pos_arr in enumerate(pos):
+
+        rot_pos_arr = np.empty((pos_arr.shape))
+
+        for col in range(0,len(pos_arr[0,:])):
+            rot_pos_arr[:,col] = rotate_point(timesteps[col], pos_arr[:,col], spin_axis, spin_velocity)
+
+        if counter == 0:
+            rotated_positions = rot_pos_arr
+        else:
+            rotated_positions = np.hstack((rotated_positions, rot_pos_arr))
+
+    # Convert the positions along the trajectory to spherical coordinates
+    r_points, theta_points, phi_points = cart2sphere(rotated_positions[0,:], rotated_positions[1,:], rotated_positions[2,:])
+
+    # Remove points outside measurement zone (i.e outside outer-bounding sphere)
+    index_feasible_positions = np.where(r_points <= radius_max) # Addition of noise to cover approximation error
+    r_points = r_points[index_feasible_positions]
+    theta_points = theta_points[index_feasible_positions]
+    phi_points = phi_points[index_feasible_positions]
+
+    # Remove points inside safety-radius (i.e inside inner-bounding sphere)
+    index_feasible_positions = np.where(r_points >= radius_min) # Addition of noise to cover approximation error
+    r_points = r_points[index_feasible_positions]
+    theta_points = theta_points[index_feasible_positions]
+    phi_points = phi_points[index_feasible_positions]
+
+    # Update and return boolean tensory given information on the new trajectory
+    if len(r_points)==0 or len(theta_points)==0 or len(phi_points)==0:
+        return bool_tensor
+    else: 
+        # Find the indices of the closest values in the meshgrid for each point using broadcasting
+        i = np.argmin(np.abs(r[:, np.newaxis] - r_points), axis=0) # indices along r axis
+        j = np.argmin(np.abs(theta[:, np.newaxis] - theta_points), axis=0) # indices along theta axis
+        k = np.argmin(np.abs(phi[:, np.newaxis] - phi_points), axis=0) # indices along phi axis
+
+        # Update tensor with information on new trajectory using advanced indexing
+        bool_tensor[i, j, k] = True
+        return bool_tensor
+
+
+def get_spherical_tensor_grid(time_step: int, radius_min: float, radius_max: float, max_velocity_scaling_factor: float) -> Union[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
```

Comment:
create_spherical_tensor_grid?
init_spherical_tensor_grid?

"get" usually means it exists already

Reply:
Fixed.